include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h

```diff
@@ -130,6 +130,26 @@
 namespace onnxruntime {
 namespace concurrency {
 
+class Profiler {
+ public:
+  Profiler() = default;
+  ~Profiler() = default;
+  ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE(Profiler);
+  using CLOCK = std::chrono::high_resolution_clock;
+  operator bool() const;
+  void Start();
+  std::string Stop();
+  void LogStart();
+  void LogEnd(std::string&& evt);
+  void LogEndAndStart(std::string&& evt);
+
+ private:
+  bool enabled_ = false;
+  std::thread::id thread_id_;
+  std::vector<std::string> events_;
```

Comment:
is the performance overhead from copying strings significant? wonder if it would be worth using an enumeration for the event types instead.
should the number of profiling events/points be unbounded?

Reply:
Abandon the string.

---
In reply to: [579575346](https://github.com/microsoft/onnxruntime/pull/6748#discussion_r579575346) [](ancestors = 579575346)